### DISCUSSION TITLE
Redirect to 404 after client side data fetch.

### DISCUSSION THREAD

**Discussion by salad-henchmen** [vercel/next.js#19872] [Help]
I have data that needn't be pre-rendered and so I'm fetching it from the client. Is there any way I can render the 404 page with the proper status code if the response is empty? I realize that a page that displays this kinda data needn't be SEO-ed but is there a method I missed to enable this?

**Comment by qnxdev** [comment]
Have you find any solution? Atleast can we set http status to 404 (like res.status in SSR) ?
